packages/protocol/contracts/L1/libs/LibProving.sol

```diff
@@ -116,50 +115,65 @@ library LibProving {
                     false
                 );
 
-                bytes memory buffer = bytes.concat(
-                    // for checking anchor tx
-                    bytes32(uint256(uint160(l1SignalService))),
-                    // for checking signalRoot
-                    bytes32(uint256(uint160(l2SignalService))),
-                    evidence.parentHash,
-                    evidence.blockHash,
-                    evidence.signalRoot
-                );
-                buffer = bytes.concat(
-                    buffer,
-                    bytes32(uint256(uint160(evidence.prover))),
-                    bytes32(uint256(evidence.meta.id)),
-                    bytes32(evidence.meta.l1Height),
-                    evidence.meta.l1Hash,
-                    evidence.meta.txListHash
-                );
-
-                instance = keccak256(buffer);
+                bytes32[11] memory inputs;
+                // for checking anchor tx
+                inputs[0] = bytes32(uint256(uint160(l1SignalService)));
+                // for checking signalRoot
+                inputs[1] = bytes32(uint256(uint160(l2SignalService)));
+                inputs[2] = evidence.parentHash;
+                inputs[3] = evidence.blockHash;
+                inputs[4] = evidence.signalRoot;
+                inputs[5] = bytes32(uint256(uint160(evidence.prover)));
+                inputs[6] = bytes32(uint256(evidence.meta.id));
+                inputs[7] = bytes32(evidence.meta.l1Height);
+                inputs[8] = evidence.meta.l1Hash;
+                inputs[9] = evidence.meta.txListHash;
+
+                // Circuits shall use this value to check anchor gas limit.
+                // Note that this value is not necessary and can be hard-coded
+                // in to the circuit code, but if we upgrade the protocol
+                // and the gas limit changes, then having it here may be handy.
+                inputs[10] = bytes32(config.anchorTxGasLimit);
```

Comment:
Brecht, since we no longer pass L2 block header to the `proveBlock` function, we don't have access to the `timestamp` and `mixhash` values on L1.

Reply:
On L2, I think we also don't need to put these values into the `inputs` array as they will be part of the block header which is available to the circuits. `chainid` is not part of th block header, thus has to be part of the input array.

Reply:
fix an issue here: https://github.com/taikoxyz/taiko-mono/pull/13336